src/Tools/CoreFx.Tools/GenerateBindingRedirect.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Text;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Task = Microsoft.Build.Utilities.Task;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateBindingRedirect : Task
+    {
+        [Required]
+        public ITaskItem[] Assemblies { get; set; }
+
+        [Required]
+        public ITaskItem[] Executables { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        [Required]
```

Comment:
nit: but we probably should make some of these properties like this one not-required, as making them Required cause older targets files to break if all the other parameters match.